protocols/api/src/main/java/org/apache/james/protocols/api/logger/ContextualLogger.java

```diff
@@ -42,7 +44,7 @@ public String get() {
         }, session.getSessionID(), logger);
     }
 
-    public ContextualLogger(Supplier<String> userSupplier, String sessionId, Logger logger) {
+    public ContextualLogger(Supplier<String> userSupplier, String sessionId, org.slf4j.Logger logger) {
```

Comment:
IMO we should not couple this.

MDC is of course desirable, but I prefer working on it separatly:
 - Java-8 PR would allow auto-closable
 - No need to delay a bugfix
 - It seems like an unrelated issue that deserves it's own ticket.